c/tskit/trees.c

```diff
@@ -4335,6 +4334,19 @@ tsk_tree_map_mutations(tsk_tree_t *self, int8_t *genotypes,
     }
     num_alleles++;
 
+    if (options & TSK_MM_FIXED_ANCESTRAL_STATE) {
+        ancestral_state = *r_ancestral_state;
+        if ((ancestral_state < 0) || (ancestral_state >= HARTIGAN_MAX_ALLELES)) {
+            ret = TSK_ERR_BAD_ANCESTRAL_STATE;
+            goto out;
+        } else if (ancestral_state >= num_alleles) {
+            num_alleles = ancestral_state;
```

Comment:
Why not ``num_alleles = ancestral_state + 1``?

Reply:
That fails with `error: conversion from ‘int’ to ‘int8_t’ {aka ‘signed char’} may change value`, as does the following (which I don't quite understand)

```
num_alleles = ancestral_state + (int8_t) 1
```

I don't mind whatever syntax as long as it compiles. Should I try `num_alleles = (int8_t) (ancestral_state + 1)`

Reply:
I don't know - there must be a way to do it. If you can't figure it out then let me know and I'll fix it.